runtime/doc/treesitter.txt

```diff
@@ -848,32 +848,55 @@ add_directive({name}, {handler}, {force})
     Handlers can set match level data by setting directly on the metadata
     object `metadata.key = value`, additionally, handlers can set node level
     data by using the capture id on the metadata table
-    `metadata[capture_id].key = value`
+    `metadata[capture_id].key = value`.
 
     Parameters: ~
-      • {name}     (string) Name of the directive, without leading #
-      • {handler}  function(match:table<string,|TSNode|>, pattern:string,
-                   bufnr:integer, predicate:string[], metadata:table)
+      • {name}     (string) Name of the directive, without leading `#`. Should
+                   end with `!`.
```

Comment:
```suggestion
      • {name}     (string) Name of the directive, without leading `#`. Must
                   end with `!`.
```

See https://github.com/neovim/neovim/blob/master/runtime/lua/vim/treesitter/query.lua#L578

Reply:
Good suggestion, I added that along with error checking.